pulp_2to3_migration/tests/functional/test_rpm_repo.py

```diff
@@ -0,0 +1,219 @@
+
+        cls.run_migration()
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Clean up the database after the set of tests is run.
+        """
+        cmd = get_psql_smash_cmd(TRUNCATE_TABLES_QUERY_BASH)
+        cls.smash_cli_client.run(cmd, sudo=True)
+        time.sleep(0.5)
+
+    def test_rpm_repo_migration(self):
+        """
+        Test that RPM repos are correctly migrated.
+
+        Check that names are migrated correctly and that the number of versions and content count is
+        correct.
+        """
+        self.assertEqual(self.rpm_repo_api.list().count, PULP_2_RPM_DATA['repositories'])
+
+        # content count in total
+        for content_type, api in self.rpm_content_apis.items():
+            with self.subTest(content_type=content_type):
+                self.assertEqual(api.list().count, PULP_2_RPM_DATA['content_total'][content_type])
+
+        for repo in self.rpm_repo_api.list().results:
+            with self.subTest(repo=repo):
+                version_count = 2 if PULP_2_RPM_DATA['content'][repo.name] else 1
+                self.assertEqual(
+                    self.rpm_repo_versions_api.list(repo.pulp_href).count, version_count
+                )
+                # content count per repo
+                for content_type, api in self.rpm_content_apis.items():
+                    with self.subTest(content_type=content_type):
+                        repo_content = api.list(repository_version=repo.latest_version_href)
+                        self.assertEqual(
+                            repo_content.count,
+                            PULP_2_RPM_DATA['content'][repo.name].get(content_type, 0)
+                        )
+
+    def test_rpm_importer_migration(self):
+        """
+        Test that RPM importers are correctly migrated.
+        """
+        self.assertEqual(self.rpm_remote_api.list().count, PULP_2_RPM_DATA['remotes'])
+        for remote in self.rpm_remote_api.list().results:
+            with self.subTest(remote=remote):
+                repo_name = '-'.join(remote.name.split('-')[1:])
+                repo_url = os.path.join(FIXTURES_BASE_URL, repo_name) + '/'
+                self.assertEqual(remote.url, repo_url)
+                self.assertEqual(remote.policy, 'on_demand')
+
+    def test_rpm_distributor_migration(self):
+        """
+        Test that RPM distributors are correctly migrated.
+        """
+        self.assertEqual(self.rpm_publication_api.list().count, PULP_2_RPM_DATA['publications'])
+        self.assertEqual(self.rpm_distribution_api.list().count, PULP_2_RPM_DATA['distributions'])
+        for dist in self.rpm_distribution_api.list().results:
+            with self.subTest(dist=dist):
+                base_path = '-'.join(dist.name.split('-')[1:])
+                self.assertEqual(dist.base_path, base_path)
+
+
+class SimplePlanMixin:
+    """A mixin class for tests to run a migration using simple plan."""
```

Comment:
How about instead of two separate Mixins, we just have:


```python
class MigrationPlanMixin:
    plan = RPM_SIMPLE_PLAN

    @classmethod
    def run_migration(cls):
        mp = cls.migration_plans_api.create({'plan': self.plan})
        mp_run_response = cls.migration_plans_api.run(mp.pulp_href, {})
        task = monitor_task(mp_run_response.task)
        monitor_task_group(task.task_group)
```

And then each migration plan can define that class variable if they want something different.  It could also be set to ```None``` and require being explicitly defined everywhere, no opinion there.

Reply:
Yessss, thank you!
Exactly what I was looking for.

Reply:
Pushed